File: packages/zynth-automation/android/src/main/java/dev/zynth/automation/ZynthAutomation.kt
package dev.zynth.automation

import android.app.Activity
import com.zynth.kit.runtime.ZynthRuntime

@Suppress("unused")
object ZynthAutomation {
    private var moduleInstance: ZynthAutomationModule? = null

    @JvmStatic
    fun initialize(activity: Activity, runtime: ZynthRuntime) {
        activity
        if (moduleInstance != null) {
            return
        }
        val module = ZynthAutomationModule(runtime)
        moduleInstance = module
        runtime.installModules(listOf(module))
    }

    @JvmStatic
    fun cleanup() {
        moduleInstance = null
    }
}

File: packages/zynth-automation/ios/ZynthAutomation.swift
import Foundation
import ZynthKit

@objc(ZynthAutomation)
public class ZynthAutomation: NSObject {
  private static var moduleInstance: ZynthAutomationModule?

  @objc public static func initialize(with runtime: ZynthRuntime) {
    guard moduleInstance == nil else {
      return
    }

    let module = ZynthAutomationModule(runtime: runtime)
    runtime.installModules([module])
    moduleInstance = module
  }

  @objc public static func cleanup() {
    moduleInstance = nil
  }
}
